coreference/bert_stuff.py

```diff
@@ -0,0 +1,175 @@
+
+# Author: Rehan and Abhijnan
+import pickle
+import os.path
+import pyhocon
+import torch.nn as nn
+import torch
+import torch.nn.functional as F
+from models import *
+import torch
+from tqdm import tqdm
+config_file_path = '../cdlm/config_pairwise_long_reg_span.json'
+config = pyhocon.ConfigFactory.parse_file(config_file_path)
+
+
+def generate_cdlm_embeddings_from_model(parallel_model,bert_sentences, device, batch_size=150):
+    """
+
+    Parameters
+    ----------
+    parallel_model: Parallel Model
+    bert_sentences: arr
+        list of sentences with mentions surrounded by <m> and </m>
+    device: torch.device
+    batch_size: int
+
+    Returns
+    -------
+        arr of vectors
+    """
+    parallel_model.eval()
+
+    # Store vectors here
+    all_vectors = []
+    
+    #vector_map_path = working_folder + f'/{men_type}_vectormaps.pkl'
+
+    loc_mentions = []
```

Comment:
what is the purpose of this?

Reply:
Purpose:  append indices of the special tokens around the mentions to this list to find overflowing tokens(that exceed 4096 token length )